src/cascadia/Remoting/Monarch.cpp

```diff
@@ -130,63 +139,239 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
             LOG_CAUGHT_EXCEPTION();
             // Remove the peasant from the list of peasants
             _peasants.erase(peasantID);
+
+            // Remove the peasant from the list of MRU windows. They're dead.
+            // They can't be the MRU anymore.
+            _clearOldMruEntries(peasantID);
             return nullptr;
         }
     }
 
+    // Method Description:
+    // - Handler for the `Peasant::WindowActivated` event. We'll make a in-proc
+    //   copy of the WindowActivatedArgs from the peasant. That way, we won't
+    //   need to worry about the origin process dying when working with the
+    //   WindowActivatedArgs.
+    // - If the peasant process dies while we're making this copy, then we'll
+    //   just log it and do nothing. We certainly don't want to track a dead
+    //   peasant.
+    // - We'll pass that copy of the WindowActivatedArgs to
+    //   _doHandleActivatePeasant, which will actually insert the
+    //   WindowActivatedArgs into the list we're using to track the most recent
+    //   peasants.
+    // Arguments:
+    // - args: the WindowActivatedArgs describing when and where the peasant was activated.
+    // Return Value:
+    // - <none>
     void Monarch::HandleActivatePeasant(const Remoting::WindowActivatedArgs& args)
     {
-        // TODO:projects/5 Use a heap/priority queue per-desktop to track which
-        // peasant was the most recent per-desktop. When we want to get the most
-        // recent of all desktops (WindowingBehavior::UseExisting), then use the
-        // most recent of all desktops.
-        const auto oldLastActiveTime = _lastActivatedTime.time_since_epoch().count();
-        const auto newLastActiveTime = args.ActivatedTime().time_since_epoch().count();
-
-        // For now, we'll just pay attention to whoever the most recent peasant
-        // was. We're not too worried about the mru peasant dying. Worst case -
-        // when the user executes a `wt -w 0`, we won't be able to find that
-        // peasant, and it'll open in a new window instead of the current one.
-        if (args.ActivatedTime() > _lastActivatedTime)
+        // Start by making a local copy of these args. It's easier for us if our
+        // tracking of these args is all in-proc. That way, the only thing that
+        // could fail due to the peasant dying is _this first copy_.
+        winrt::com_ptr<implementation::WindowActivatedArgs> localArgs{ nullptr };
+        try
+        {
+            localArgs = winrt::make_self<implementation::WindowActivatedArgs>(args.PeasantID(),

```

Comment:
nit: I'd expect this to act more like a copy constructor

Reply:
(take in a projected type, unbox it internally. that way, it's a one-stop shop for "where do I add new fields to copy?")